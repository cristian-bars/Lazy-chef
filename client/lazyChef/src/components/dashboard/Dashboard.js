import React from 'react';
import {View, Image, TouchableOpacity, Text} from 'react-native';
import styles from './styles';
import generalStyles from '../../../styles';
import Login from '../login/Login';

const Dashboard = ({navigation}) => {
  return (
    <View style={generalStyles.container}>
      <Image
        style={styles.logo}
        source={{
          uri: 'https://i.ibb.co/6vTqDm8/Logo-Trabajo-Final-Patxi.png',
        }}
      />
      <TouchableOpacity style={generalStyles.button}>
        <Text style={generalStyles.baseText}>Acceder</Text>
      </TouchableOpacity>
      <TouchableOpacity
        style={styles.touchRegister}
        onPress={() => navigation.navigate('Register')}>
        <Text style={styles.register}>Crear una cuenta</Text>
      </TouchableOpacity>
    </View>
  );
};

export default Dashboard;
