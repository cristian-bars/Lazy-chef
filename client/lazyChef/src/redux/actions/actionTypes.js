export default {
  LOAD_RECIPES: 'LOAD_RECIPES',
  DELETE_RECIPE: 'DELETE_RECIPE',
  ADD_RECIPE: 'ADD_RECIPE',
  UPDATE_RECIPE: 'UPDATE_RECIPE',
  LOAD_RECIPE: 'LOAD_RECIPE',

  LOAD_USERS: 'LOAD_USERS',
  DELETE_USER: 'DELETE_USER',
  ADD_USER: 'ADD_USER',
  UPDATE_USER: 'UPDATE_USER',
  LOAD_USER: 'LOAD_USER',
};
