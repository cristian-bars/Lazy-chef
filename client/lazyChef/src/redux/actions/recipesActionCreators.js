import actionTypes from './actionTypes';
import axios from 'axios';
import {env} from '../../../.env.js';

const url = env.REACT_APP_PORT;

export function loadRecipes() {
  return async dispatch => {
    try {
      const {data} = await axios(`${url}/recipes`);
      dispatch({
        type: actionTypes.LOAD_RECIPES,
        recipes: data,
      });
    } catch (error) {
      dispatch({
        type: 'LOAD_RECIPES_ERROR',
      });
    }
  };
}

export function addRecipe(recipe) {
  return async dispatch => {
    try {
      const {data} = await axios.post(`${url}/recipes`, recipe);
      dispatch({
        type: actionTypes.ADD_RECIPE,
        recipe: data,
      });
    } catch (error) {
      dispatch({
        type: 'ADD_RECIPE_ERROR',
      });
    }
  };
}

export function deleteRecipe(recipeId) {
  return async dispatch => {
    try {
      await axios.delete(`${url}/recipes/${recipeId}`);
      dispatch({
        type: actionTypes.DELETE_RECIPE,
        recipeId,
      });
    } catch (error) {
      dispatch({
        type: 'DELETE_RECIPE_ERROR',
      });
    }
  };
}

export function updateRecipe(recipe) {
  return async dispatch => {
    try {
      const {data} = await axios.post(`${url}/recipes/${recipe._id}`, recipe);
      dispatch({
        type: actionTypes.UPDATE_RECIPE,
        recipe: data,
      });
    } catch (error) {
      dispatch({
        type: 'UPDATE_RECIPE_ERROR',
      });
    }
  };
}

export function getRecipeById(recipeId) {
  return async dispatch => {
    try {
      const {data} = await axios(`${url}/recipes/${recipeId}`);
      dispatch({
        type: actionTypes.LOAD_RECIPE,
        recipe: data,
      });
    } catch (error) {
      dispatch({
        type: 'LOAD_RECIPE_ERROR',
      });
    }
  };
}
