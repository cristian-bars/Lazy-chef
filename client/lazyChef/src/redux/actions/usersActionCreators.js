import actionTypes from './actionTypes';
import axios from 'axios';

//const url = 'http://localhost/users';
const url = 'http://192.168.0.29:2022';
//const url = process.env.REACT_APP_PORT;

export function loadUsers() {
  return async dispatch => {
    try {
      const {data} = await axios(url);
      dispatch({
        type: actionTypes.LOAD_USERS,
        users: data,
      });
    } catch (error) {
      dispatch({
        type: 'LOAD_USERS_ERROR',
      });
    }
  };
}

export function addUsers(user) {
  return {
    type: actionTypes.ADD_USER,
    user,
  };
}

export function deleteUser(userId) {
  return async dispatch => {
    await axios.delete(`${url}/${userId.id}`);
    dispatch({
      type: actionTypes.DELETE_USER,
      userId,
    });
  };
}

export function updateUser(user) {
  return {
    type: actionTypes.UPDATE_USER,
    user,
  };
}

export function loadUser(user) {
  return {
    type: actionTypes.LOAD_USER,
    user,
  };
}

export function getUserById(user) {
  return async dispatch => {
    try {
      const {data} = await axios.post(`${url}/login`, user);
      dispatch({
        type: actionTypes.LOAD_USER,
        userToken: data,
      });
    } catch (error) {
      dispatch({
        type: 'LOAD_USERS_ERROR',
      });
    }
  };
}
