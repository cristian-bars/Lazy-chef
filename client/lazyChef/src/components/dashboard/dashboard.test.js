import 'react-native';
import React from 'react';
import Dashboard from './index';
import thunk from 'redux-thunk';
import Store from 'redux-mock-store';
import * as actions from '../../redux/actions/recipesActionCreators';
import {Provider} from 'react-redux';
import {render, fireEvent} from '@testing-library/react-native';

describe('Given a Login component', () => {
  const mockStore = Store([thunk]);
  let navigation;
  let myStore;
  let route;
  beforeEach(() => {
    jest.spyOn(actions, 'loadRecipes').mockReturnValueOnce({type: ''});
    myStore = mockStore({
      recipes: [
        {
          title: 'macarrones',
          description: '1a',
          image: ['a'],
          recipeIngredient: ['b'],
        },
        {
          title: 'Queso',
          description: '2a',
          image: ['a'],
          recipeIngredient: ['b'],
        },
      ],
    });
    navigation = {
      navigate: jest.fn(),
    };
    route = {
      recipes: '',
    };
  });

  test('renders correctly', () => {
    const wrapper = render(
      <Provider store={myStore}>
        <Dashboard />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  // test('Then actions.loadRecipes is invoked', () => {
  //   const recipes = [];
  //   render(
  //     <Provider store={myStore}>
  //       <Dashboard route={route} />
  //     </Provider>,
  //   );
  //   expect(actions.loadRecipes).toHaveBeenCalled();
  // });
});
