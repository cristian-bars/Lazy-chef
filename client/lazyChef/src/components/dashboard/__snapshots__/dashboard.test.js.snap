// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Login component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#D3E0FF",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderColor": "#397ED0",
          "borderRadius": 25,
          "borderWidth": 2,
          "display": "flex",
          "flexDirection": "row",
          "fontSize": 25,
          "height": 50,
          "margin": 15,
          "marginTop": 20,
          "paddingLeft": 15,
          "textAlign": "center",
          "width": 360,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/img/search.png",
          }
        }
        style={
          Object {
            "height": 30,
            "opacity": 0.7,
            "width": 30,
          }
        }
      />
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        placeholder="Buscar"
        rejectResponderTermination={true}
        style={
          Object {
            "fontSize": 20,
            "paddingLeft": 20,
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "flexDirection": "column",
          "flexWrap": "wrap",
          "margin": 10,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "backgroundColor": "white",
            "color": "#FF914D",
            "fontSize": 30,
          }
        }
      >
        Ingredientes
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderRadius": 5,
              "borderWidth": 1,
              "marginHorizontal": 10,
              "marginVertical": 10,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text>
            Huevo
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderRadius": 5,
              "borderWidth": 1,
              "marginHorizontal": 10,
              "marginVertical": 10,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text>
            Tomate
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderRadius": 5,
              "borderWidth": 1,
              "marginHorizontal": 10,
              "marginVertical": 10,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text>
            Jud√≠as
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderRadius": 5,
              "borderWidth": 1,
              "marginHorizontal": 10,
              "marginVertical": 10,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text>
            Arroz
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View>
    <RCTScrollView
      data={
        Array [
          Object {
            "description": "1a",
            "image": Array [
              "a",
            ],
            "recipeIngredient": Array [
              "b",
            ],
            "title": "macarrones",
          },
          Object {
            "description": "2a",
            "image": Array [
              "a",
            ],
            "recipeIngredient": Array [
              "b",
            ],
            "title": "Queso",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "white",
                "flexDirection": "column",
                "margin": 10,
                "opacity": 1,
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "a",
                }
              }
              style={
                Object {
                  "height": 150,
                }
              }
            />
            <Text
              style={
                Object {
                  "flex": 1,
                  "flexWrap": "wrap",
                  "fontSize": 20,
                  "padding": 10,
                }
              }
            >
              macarrones
            </Text>
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-around",
                }
              }
            >
              <View>
                <Image
                  source={
                    Object {
                      "testUri": "../../../src/img/Ingredientes.jpg",
                    }
                  }
                  style={
                    Object {
                      "height": 50,
                      "width": 50,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "flexWrap": "wrap",
                      "fontSize": 15,
                      "padding": 10,
                    }
                  }
                >
                  1
                </Text>
              </View>
              <View>
                <Image
                  source={
                    Object {
                      "testUri": "../../../src/img/herramientas-y-utensilios.png",
                    }
                  }
                  style={
                    Object {
                      "height": 50,
                      "width": 50,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "flexWrap": "wrap",
                      "fontSize": 15,
                      "padding": 10,
                    }
                  }
                />
              </View>
              <View>
                <Image
                  source={
                    Object {
                      "testUri": "../../../src/img/duration.png",
                    }
                  }
                  style={
                    Object {
                      "height": 50,
                      "width": 50,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "flexWrap": "wrap",
                      "fontSize": 15,
                      "padding": 10,
                    }
                  }
                />
              </View>
            </View>
            <Text
              style={
                Object {
                  "flexWrap": "wrap",
                  "fontSize": 15,
                  "padding": 10,
                }
              }
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "white",
                "flexDirection": "column",
                "margin": 10,
                "opacity": 1,
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "a",
                }
              }
              style={
                Object {
                  "height": 150,
                }
              }
            />
            <Text
              style={
                Object {
                  "flex": 1,
                  "flexWrap": "wrap",
                  "fontSize": 20,
                  "padding": 10,
                }
              }
            >
              Queso
            </Text>
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-around",
                }
              }
            >
              <View>
                <Image
                  source={
                    Object {
                      "testUri": "../../../src/img/Ingredientes.jpg",
                    }
                  }
                  style={
                    Object {
                      "height": 50,
                      "width": 50,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "flexWrap": "wrap",
                      "fontSize": 15,
                      "padding": 10,
                    }
                  }
                >
                  1
                </Text>
              </View>
              <View>
                <Image
                  source={
                    Object {
                      "testUri": "../../../src/img/herramientas-y-utensilios.png",
                    }
                  }
                  style={
                    Object {
                      "height": 50,
                      "width": 50,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "flexWrap": "wrap",
                      "fontSize": 15,
                      "padding": 10,
                    }
                  }
                />
              </View>
              <View>
                <Image
                  source={
                    Object {
                      "testUri": "../../../src/img/duration.png",
                    }
                  }
                  style={
                    Object {
                      "height": 50,
                      "width": 50,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "flexWrap": "wrap",
                      "fontSize": 15,
                      "padding": 10,
                    }
                  }
                />
              </View>
            </View>
            <Text
              style={
                Object {
                  "flexWrap": "wrap",
                  "fontSize": 15,
                  "padding": 10,
                }
              }
            />
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
